lib/attributor/types/model.rb

```diff
@@ -32,97 +48,40 @@ def self.define_accessors(name)
       self.define_writer(name)
     end
 
-
     def self.define_reader(name)
       module_eval <<-RUBY, __FILE__, __LINE__ + 1
         def #{name}
-          return @attributes[:#{name}] if @attributes.has_key?(:#{name})
-
-          @attributes[:#{name}] = begin
-            if (proc = @lazy_attributes.delete :#{name})
-              if proc.arity > 0
-                proc.call(self)
-              else
-                proc.call
-              end
-            end
-          end
+          @contents[:#{name}]
         end
       RUBY
     end
 
 
     def self.define_writer(name)
-      attribute = self.attributes[name]
       context = ["assignment","of(#{name})"].freeze
-      # note: paradoxically, using define_method ends up being faster for the writer
-      #       attribute is captured by the block, saving us from having to retrieve it from
-      #       the class's attributes hash on each write.
       module_eval do
         define_method(name.to_s + "=") do |value|
-          # TODO: what type of context do we report with unscoped assignments?
-          #  => for now this would report "assignment.of(field_name)" is that good?
-          @attributes[name] = attribute.load(value,context)
+          self.set(name, value, context: context)
         end
       end
     end
 
-
     def self.describe(shallow=false)
       hash = super
-
-      # Spit attributes if it's the root or if it's an anonymous structures
-      if ( !shallow || self.name == nil) && self.attributes
-        hash[:attributes] = self.attributes.each_with_object({}) do |(sub_name, sub_attribute), sub_attributes|
-          sub_attributes[sub_name] = sub_attribute.describe(true)
-        end
-      end
-
+      hash[:attributes] = hash.delete :keys
       hash
     end
 
 
-    def self.example(context=nil, **values)
-      result = self.new
-
-      context = case context
-      when nil
-        ["#{self.name}-#{result.object_id.to_s}"]
-      when ::String
-        [context]
-      else
-        context
-      end
-
-      example_depth = context.size
-
-      self.attributes.each do |sub_attribute_name,sub_attribute|
-        if sub_attribute.attributes
-           # TODO: add option to raise an exception in this case?
-           next if example_depth > MAX_EXAMPLE_DEPTH
-        end
-
-        sub_context = self.generate_subcontext(context,sub_attribute_name)
-
-        result.lazy_attributes[sub_attribute_name] = Proc.new do
-          value = values.fetch(sub_attribute_name) do
-            sub_attribute.example(sub_context, parent: result)
-          end
-
-          sub_attribute.load(value,sub_context)
-        end
+    # an example, but with forced evaluation of lazy attributes
+    def self.example!(context=nil, **values)
```

Comment:
Yes, in one of the collection specs at least, and possibly will in others. 

Although, it's probably not needed with the change to the contents accessor.